Tests/Shared/Core/UnitTestBase.swift

```diff
@@ -108,29 +108,42 @@ open class UnitTestBase<Component>: XCTestCase {
         diGraphShared.reset()
     }
 
+    // All data that the SDK writes, delete it here so each test function has a clean environment to run and does not depend on the result of the previous test.
     open func deleteAllPersistentData() {
-        var expectations: [XCTestExpectation] = []
-
-        let resetEventBusExpectation = XCTestExpectation(description: "reset EventBus to initial state")
-        expectations.append(resetEventBusExpectation)
-        Task {
-            await diGraphShared.eventBusHandler.reset()
-            resetEventBusExpectation.fulfill()
-        }
-
-        // The SDK does not use `UserDefaults.standard`, but in case a test needs to,
-        // let's delete the data for each test.
-        UserDefaults.standard.deleteAll()
+        deleteAllFiles()
+        deleteKeyValueStoredData()
+    }
 
-        // delete key value data that belongs to shared storage.
+    // Deletes all key/value storage pairs that the SDK could save
+    private func deleteKeyValueStoredData() {
         diGraphShared.sharedKeyValueStorage.deleteAll()
+    }
+
+    private func deleteAllFiles() {
+        let fileManager = FileManager.default
+
+        let deleteFromSearchPath: (FileManager.SearchPathDirectory) -> Void = { path in
+            // OK to use try! here as we want tests to crash if for some reason we are not able to delete files from the
+            // device.
+            // if files do not get deleted between tests, we could have false positive tests.
+            // swiftlint:disable:next force_try
+            let pathUrl = try! fileManager.url(for: path, in: .userDomainMask, appropriateFor: nil, create: false)
```

Comment:
even though we run tests in a controlled environment, I would have hoped we had a dedicated directory for testing and we just cleaned it. 

currently the files were being deleted using 

```
 /// Resets the EventBus to initial state by removing all observers and stored events.
    public func reset() async {
        await eventStorage.removeAll()
        await eventBus.removeAllObservers()
    }
    
    // and removeAll looks like this
    func removeAll() async {
        try? fileManager.removeItem(at: baseDirectory)
    }
```

Reply:
I don't understand the question, sorry.